agent/job_runner.go

```diff
@@ -173,11 +173,11 @@ func (r *JobRunner) Kill() error {
 	defer r.killLock.Unlock()
 
 	if !r.cancelled {
-		logger.Info("Canceling job %s", r.Job.ID)
+		logger.Info("Canceling job %s with a grace period of %ds", r.Job.ID, r.AgentConfiguration.CancelGracePeriod)
 		r.cancelled = true
 
 		if r.process != nil {
-			r.process.Kill()
+			r.process.Kill(time.Second * time.Duration(r.AgentConfiguration.CancelGracePeriod))
```

Comment:
Does this need to be Seconds * Duration, or could it just be Second * int, like it was previously?

Reply:
time.Second is of type `time.Duration`, so the `time.Duration(...)` is just a cast to match the types.